enclave/core/sgx/report.c

```diff
@@ -314,7 +314,7 @@ oe_result_t oe_get_report_v1(
     header->report_type = (flags & OE_REPORT_FLAGS_REMOTE_ATTESTATION)
                               ? OE_REPORT_TYPE_SGX_REMOTE
                               : OE_REPORT_TYPE_SGX_LOCAL;
-    header->report_size = *report_buffer_size;
+    header->report_size = (uint64_t)*report_buffer_size;
```

Comment:
Optional: consider removing this cast. Either **sizeof(size_t)** <= **sizeof(uint64_t)**, in which case you don't need a cast, or **sizeof(size_t)** > **sizeof(uint64_t)**, in which case the cast is erroneous. The latter case is impossible on a 64-bit platform but will be a real problem with 128-bit platforms (future).

Reply:
Good point. I will remove this unnecessary casting.
Thanks for the great feedback.